docs/node/migrations/zero-downtime-migrations/migration-proposal.mdx

```diff
@@ -0,0 +1,57 @@
+---
+sidebar_position: 1
+sidebar_label: "Migration Proposal"
+id: migration-proposal
+title: Migration Proposal
+description: "Network governance proposal to trigger a network migration"
+slug: /node/migrations/zero-downtime-migrations/migration-proposal
+---
+
+A network migration is proposed by a validator using a special network governance resolution. The migration proposal includes the activation period, duration, and the chain ID of the new network as shown below.
+
+- `ActivationPeriod`: The amount of blocks before the migration is activated. It starts after the migration is approved via the voting system. The intention is to allow validators to prepare for the migration.
+- `Duration`: The amount of blocks the migration will take to complete.
+- `ChainID`: The new chain ID that the network will migrate to. A new chain ID should always be used for a new network to avoid cross-network replay attacks.
+- `Timestamp`: The time at which the migration was proposed.
+
+The Migration proposal needs to be approved by atleast 2/3rd of the validators to start the migration process. The migration proposal will expire if the supermajority of the validators do not approve the proposal within `100800` blocks (roughly 7 days with a 6sec block interval). Once the migration proposal is approved, the network moves into migration mode after the specified activation period.
+
+## Submit Migration Proposal
+
+A validator operator can submit a migration proposal to upgrade the network using the `kwil-admin migrate propose` command as shown below. Refer to the [kwil-admin migrate](/docs/ref/kwil-admin/migration/propose) documentation for more details.
+
+```bash
+kwil-admin migrate propose --activation-period <activationPeriod> --duration <migrationDuration> --chain-id <newChainID> --rpcserver /tmp/kwil.sock
+```
+
+:::note
+An off-chain communication is required among the validator operators to decide on the details of the migration proposal. One of the validators should submit the proposal to the network for approval. Only one proposal can be applied to the network during the lifetime of the network. Once the proposal is approved, all the other migration proposals will be rejected from the network and no new proposals can be submitted or approved.
```

Comment:
Does this mean that a chain can only have one migration proposal over its entire lifetime? For example, TSN uses zero-downtime migrations to go from kwil v0.9 to kwil v0.10. Does that mean it can't use zero downtime migrations to go from v0.10 to v0.11?

Reply:
I have a feeling the actual behavior is 1 network migration per chain-id. If that is the case, this note should say so.